decompiler/pipeline/controlflowanalysis/variable_name_generation.py

```diff
@@ -0,0 +1,123 @@
+import re
+from enum import Enum
+from typing import List, Optional
+
+from decompiler.pipeline.stage import PipelineStage
+from decompiler.structures.ast.ast_nodes import CaseNode, CodeNode, ConditionNode, LoopNode, SwitchNode
+from decompiler.structures.ast.syntaxtree import AbstractSyntaxTree
+from decompiler.structures.pseudo import CustomType, DataflowObject, Float, GlobalVariable, Integer, Pointer, Type, Variable
+from decompiler.task import DecompilerTask
+
+
+def _get_var_counter(var_name: str) -> Optional[str]:
+    if counter := re.match(r".*?([0-9]+)$", var_name):
+        return counter.group(1)
+    return None
```

Comment:
I wonder if there is merit in perserving the old variable numbers.
However, I need a short docstring here, maybe something among the lines of:
```python
"""Parse the index-suffix from the given variable name."""
```

Reply:
Docstring done, but what do you mean by "preserving the old variable number"? 
The new one will get the same as the old one.

Reply:
This is what I referr to as "preserving" ;)